consensus/src/persistent_liveness_storage.rs

```diff
@@ -23,8 +24,13 @@ use std::{cmp::max, collections::HashSet, sync::Arc};
 /// Blocks persisted are proposed but not yet committed.  The committed state is persisted
 /// via StateComputer.
 pub trait PersistentLivenessStorage: Send + Sync {
-    /// Persist the blocks and quorum certs into storage atomically.
-    fn save_tree(&self, blocks: Vec<Block>, quorum_certs: Vec<QuorumCert>) -> Result<()>;
+    /// Persist the blocks, quorum certs and highest ordered cert into storage atomically.
+    fn save_tree(
+        &self,
+        blocks: Vec<Block>,
+        quorum_certs: Vec<QuorumCert>,
+        highest_ordered_cert: Option<WrappedLedgerInfo>,
```

Comment:
it seems better to have a separate api for this

Reply:
I originally had a separate API. Then I merged it for atomicity. If I have a separate API, then there could be a scenario where blocks and quorum certs are committed to storage, and then validator crashes and didn't commit highest_ordered_cert to storage.